frontend/src/lib/hooks/useAddressBook/index.ts

```diff
@@ -3,18 +3,44 @@ import {
   usePersistentState,
 } from '../../../state/persistentState';
 import { useStateCorrector } from '../../ui/state/useStateCorrector';
+import { isValidAddress } from './utils/isValidAddress';
+
+export type AddressBookEntry = {
+  address: string;
+  coinSymbol: string;
+  title: string;
+};
 
 export const useAddressBook = () => {
   const [addresses, setAddresses] = useStateCorrector(
-    usePersistentState<string[]>(PersistentStateKey.AddressBook, []),
-    addresses => addresses.filter(Boolean)
+    usePersistentState<AddressBookEntry[]>(PersistentStateKey.AddressBook, []),
```

Comment:
We have AddressBook setup to save to database , refer to `SaveAddressBookItem` in `Store.d.ts`